collections/_article/whats-new-in-csharp-for-godot-4-0.md

```diff
@@ -0,0 +1,357 @@
+
+We are looking into ways to support extending our source generators to allow custom user attributes to affect scripts, so keep an eye out for that in the future.
+
+In Godot 4.0, GDScript added a bunch of new export annotations. However, we didn't have time to add equivalent attributes in C#. All those annotations can be replicated with the `[Export]` attribute using the right combination of `PropertyHint` and `HintString` values, until we implement these convenient attributes in a future version of Godot 4.x.
+
+As for [custom resources](#global-classes), we were unable to finish this work on time for 4.0, but we're actively working on it.
+
+#### Relevant links
+
+- Ability to export Node types instead of just NodePaths ([GH-1048](https://github.com/godotengine/godot-proposals/issues/1048)).
+- Add ability to export Node pointers as NodePaths ([GH-62185](https://github.com/godotengine/godot/pull/62185)).
+- C#: Enable exporting nodes to the inspector ([GH-62789](https://github.com/godotengine/godot/pull/62789)).
+- Add grouping annotations for class properties in GDScript ([GH-62707](https://github.com/godotengine/godot/pull/62707)).
+- C#: Add an easy and accessible way of organizing export variables ([GH-3451](https://github.com/godotengine/godot-proposals/issues/3451)).
+- C#: Add grouping attributes for properties ([GH-64742](https://github.com/godotengine/godot/pull/64742)).
+- C#: Preserve order of exported fields/categories ([GH-64852](https://github.com/godotengine/godot/pull/64852)).
+- Support explicit values in flag properties, add C# flags support ([GH-59327](https://github.com/godotengine/godot/pull/59327)).
+
+## Improved C# documentation
+
+We have updated the [C# differences documentation](https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_differences.html) to include tables with the C# equivalent of common GDScript API. Sometimes an equivalent in the Godot C# API does not exist because there are existing methods in the .NET Base Class Library that we would like you to use instead, this reduces the maintainability burden since we don't have to provide that API in Godot and helps users discover C# API that they can use in non-Godot projects.
+
+The new equivalence tables go into as much detail about what's available, and will hopefully be a great tool for developers looking to port GDScript code to C#. Let us know if this can be improved!
```

Comment:
```suggestion
A lot of Godot tutorials and tips you can find online contain samples of GDScript code. To help you with interpreting them into C# equivalents the official documentation has a [C# differences article](https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_differences.html), which has also been updated. It now features a complete set of tables with the C# equivalent for every common GDScript API. Sometimes a Godot C# equivalent does not exist because there are more suitable methods in the .NET Base Class Library.

The new equivalence tables try to encompass every possible case, and should be a great tool for developers looking to port GDScript code to C#. Let us know if this can be improved!
```